### PR TITLE
Feature/pipe 166

### PR DESCRIPTION
**Pull request by Liametc**
### Changed
- To fix issue where houdini presets are applied before the node handlers, there is now a check to prevent any duplicate parameters being added. Any duplicate is logged by shotgun for debugging should things not be as expected for the user.

### Added
- We can now set default parameters for the nodes within the shotgun settings using the `init_node_values` key.

### CODE REVIEW

File: python/tk_houdini/utils/parm_template_wrappers.py [hunk 1]
```diff
@@ -93,25 +163,60 @@ def index_of_template(self, template_name):
...
+    def extend(self, parms):
+        """
+        Add multiple templates to the child templates list..
+
+        :param list(Parm)) parms: The list of Parm
+            instances to add.
+        """
+        for parm in parms:
+            self.append(parm)
+
```
Comment (j0yu) [inline python/tk_houdini/utils/parm_template_wrappers.py:177]: If you fancy, you can also implement [\_\_iadd\_\_](https://docs.python.org/3/reference/datamodel.html#object.__iadd__) to be the same as `extend()`, allowing you to have the option to `folder += [param1, param2, param3]`

```suggestion

    __iadd__ = extend
```

File: python/tk_houdini/utils/parm_template_wrappers.py [hunk 2]
```diff
@@ -93,25 +163,60 @@ def index_of_template(self, template_name):
...
             instances to add.
         """
-        template_names = (template.name() for template in templates)
-        child_templates = (Parm.build_parm(template) for template in templates)
-        self.__child_names.extend(template_names)
-        self.__children.extend(child_templates)
+        child_templates = (
+            Parm.build_parm(template, parent=self) for template in templates
+        )
+        self.extend(templates)
```
Comment (j0yu) [inline python/tk_houdini/utils/parm_template_wrappers.py]: Shouldn't this be:
```python
        self.extend(child_templates)
```

You could also just shorten it to:
```suggestion
        self.extend(
            Parm.build_parm(template, parent=self) for template in templates
        )
```

File: python/tk_houdini/utils/parm_template_wrappers.py [hunk 3]
```diff
@@ -21,27 +24,31 @@ class Parm(object):
     """
 
     @classmethod
-    def build_parm(cls, template):
+    def build_parm(cls, template, parent=None):
```
Comment (j0yu) [inline python/tk_houdini/utils/parm_template_wrappers.py]: IGNORABLE: Damn, I should have caught this much earlier but this method might be more explicitly named as
```python
def from_template(cls, template, parent=None):
```

**Review by j0yu — COMMENTED** [review]
Nice stuff

File: python/tk_houdini/utils/parm_template_wrappers.py (hunk 3 above)
Comment (Liametc) [reply python/tk_houdini/utils/parm_template_wrappers.py]: yeah, that is a good idea

File: python/tk_houdini/utils/parm_template_wrappers.py (hunk 1 above)
Comment (Liametc) [reply python/tk_houdini/utils/parm_template_wrappers.py:177]: I might leave it for now. It's not really used anywhere

Comment (Liametc) [reply python/tk_houdini/utils/parm_template_wrappers.py:177]: ah f*** it lets just put it in

**Review by j0yu — APPROVED** [review]

**Pull request merged**